tests/unit/casmutils/structure_tools.cpp

```diff
@@ -34,18 +41,22 @@ class StructureToolsTest : public testing::Test
         primitive_fcc_Ni_ptr = std::make_unique<Structure>(Structure::from_poscar(primitive_path));
         deformed_conventional_fcc_Ni_ptr =
             std::make_unique<Structure>(Structure::from_poscar(deformed_conventional_path));
+        strained_conventional_fcc_Ni_ptr =
+            std::make_unique<Structure>(Structure::from_poscar(strained_conventional_path));
     }
 
     // Use unique pointers because Structure has no default constructor
     std::unique_ptr<Structure> cubic_Ni_struc_ptr;
     std::unique_ptr<Structure> conventional_fcc_Ni_ptr;
     std::unique_ptr<Structure> nonniggli_conventional_fcc_Ni_ptr;
     std::unique_ptr<Structure> deformed_conventional_fcc_Ni_ptr;
+    std::unique_ptr<Structure> strained_conventional_fcc_Ni_ptr;
     std::unique_ptr<Structure> primitive_fcc_Ni_ptr;
 
     double tol = 1e-5;
 
     Eigen::Matrix3d deformation;
+    std::vector<double> GL_unrolled_strain;
```

Comment:
You can simplify the tests by directly storing the unrolled tensor as `Eigen::VectorXd`:
```
Eigen::VectorXd unrolled_GL_tensor;
```
Then in `SetUp()`:
```
deformation << 1.0, 0.02, 0.02, 0.0, 1.0, 0.0, 0.0, 0.0, 1.05;

Eigen::Matrix3d GL_tensor = (deformation * deformation - Eigen::Matrix3d::Identity()) / 2;
unrolled_GL_tensor << GL_tensor(0, 0), GL_tensor(1, 1), GL_tensor(2, 2), 
                      GL_tensor(1, 2), GL_tensor(0, 2), GL_tensor(0, 1);
```
and in the `TEST_F` you can just directly call the value from the fixture without the `Eigen::Map` conversion.